src/databricks/labs/ucx/assessment/dashboards.py

```diff
@@ -0,0 +1,370 @@
+            catalog=catalog or cls.catalog,
+            schema=schema or cls.schema,
+            tags=query.tags or [],
+        )
+
+    @classmethod
+    def from_lakeview_dataset(cls, dataset: Dataset, *, parent: str | None = None) -> Query:
+        """Create query from a :class:Dataset"""
+        return cls(
+            id=dataset.name,
+            name=dataset.display_name or cls.name,
+            parent=parent or cls.parent,
+            query=dataset.query,
+        )
+
+
+@dataclass
+class Dashboard:
+    """UCX representation of a dashboard."""
+
+    id: str
+    """The ID for this dashboard."""
+
+    name: str = "UNKNOWN"
+    """The title of the dashboard that appears in list views and at the top of the dashboard page."""
+
+    parent: str = "ORPHAN"
+    """The identifier of the workspace folder containing the object."""
+
+    query_ids: list[str] = field(default_factory=list)
+    """The IDs of the queries referenced by this dashboard."""
+
+    tags: list[str] = field(default_factory=list)
+    """The tags set on this dashboard."""
+
+    @classmethod
+    def from_sdk_redash_dashboard(cls, dashboard: SdkRedashDashboard) -> Dashboard:
+        query_ids = []
+        for widget in dashboard.widgets or []:
+            if widget.visualization is None:
+                continue
+            if widget.visualization.query is None:
+                continue
+            if widget.visualization.query.id is None:
+                continue
+            query_ids.append(widget.visualization.query.id)
+        return cls(
+            id=dashboard.id or cls.id,
+            name=dashboard.name or cls.name,
+            parent=dashboard.parent or cls.parent,
+            query_ids=query_ids,
+            tags=dashboard.tags or [],
+        )
+
+    @classmethod
+    def from_sdk_lakeview_dashboard(cls, dashboard: SdkLakeviewDashboard) -> Dashboard:
+        assert dashboard.dashboard_id
+        lsql_dashboard = _convert_sdk_to_lsql_lakeview_dashboard(dashboard)
+        query_ids = [dataset.name for dataset in lsql_dashboard.datasets]
+        return cls(
+            id=dashboard.dashboard_id,
+            name=dashboard.display_name or cls.name,
+            parent=dashboard.parent_path or cls.parent,
+            query_ids=query_ids,
+        )
+
+
+class RedashDashboardCrawler(CrawlerBase[Dashboard]):
+    """Crawler for Redash dashboards."""
+
+    def __init__(
+        self,
+        ws: WorkspaceClient,
+        sql_backend: SqlBackend,
+        schema: str,
+        *,
+        include_dashboard_ids: list[str] | None = None,
+        include_query_ids: list[str] | None = None,
+        debug_listing_upper_limit: int | None = None,
+    ):
+        super().__init__(sql_backend, "hive_metastore", schema, "redash_dashboards", Dashboard)
+        self._ws = ws
+        self._include_dashboard_ids = include_dashboard_ids or []
+        self._include_query_ids = include_query_ids or []
```

Comment:
I think this precludes us from configuring it with an empty list (`[]`), the obvious way to prevent it from crawling any dashboards/queries.

It would be cleaner to leave `None` as meaning "crawl everything".